tests/lean/pp_numeral_types.lean.expected.out

```diff
@@ -1,5 +1,11 @@
 (17 : fin 22) : fin 22
+(17 : fin (22 : ℕ)) : fin (22 : ℕ)
+(3 : ℕ) : ℕ
 (2 : ℕ) + (1 : ℕ) = (3 : ℕ) : Prop
 (17 : fin (22 : ℕ)) : fin (22 : ℕ)
 (3 : ℕ) : ℕ
-(3 : nat) : ℕ
+(3 : ℕ) : ℕ
+@bit1.{0} nat nat.has_one nat.has_add (@has_one.one.{0} nat nat.has_one) : nat
+nat.succ (nat.succ (nat.succ nat.zero)) : nat
+3 : ℕ
+1.succ.succ : ℕ
```

Comment:
This is not ideal, since such an expression doesn't round-trip.  I'm not sure if there's a good way to fix this.

Maybe it makes sense to scale back this PR to just changing the block of code at line 2042 of src/frontends/lean/pp.cpp and not adding the `pp.nat_numerals` option.